CMakeLists.txt

```diff
@@ -22,6 +22,18 @@ endif()
 set(CMAKE_MODULE_PATH ${CMAKE_CURRENT_SOURCE_DIR}/cmake ${CMAKE_MODULE_PATH})
 include(ExternalProject)
 
+# ---- C++ Modules Support (optional) ----
+option(NLOHMANN_JSON_BUILD_MODULES "Build C++ modules support" OFF)
+
+if(NLOHMANN_JSON_BUILD_MODULES)
+    if(CMAKE_VERSION VERSION_GREATER_EQUAL 3.28)
+        message(STATUS "Building nlohmann-json C++ modules (CMake ${CMAKE_VERSION} supports modules)")
```

Comment:
I don't think repeating the user's CMake version in this message is helpful.

Reply:
Additionally, maybe the message should be `Building nlohmann.json C++ module` to make it easier to understand that `import nlohmann.json;` is required to use it.